src/main/java/org/prebid/server/analytics/AnalyticsReporterDelegator.java

```diff
@@ -168,4 +181,43 @@ private static ObjectNode prepareAnalytics(ObjectNode analytics, String adapterN
 
         return !analyticsNodeCopy.isEmpty() ? analyticsNodeCopy : null;
     }
+
+    private <T> void processEventWithResult(T event, AnalyticsReporter analyticsReporter) {
+        final String analyticsCode = analyticsReporter.name();
+        analyticsReporter.processEvent(event)
+                .compose(s -> processSuccess(s, analyticsCode))
+                .recover(s -> processFail(s, event, analyticsCode));
+    }
+
+    private <T> Future<Void> processSuccess(T event, String analyticsCode) {
+        updateMetricsByEventType(event, analyticsCode, MetricName.ok);
+        return Future.succeededFuture();
+    }
+
+    private <T> Future<Void> processFail(Throwable exception, T event, String analyticsCode) {
+        if (exception instanceof TimeoutException || exception instanceof ConnectTimeoutException) {
+            updateMetricsByEventType(event, analyticsCode, MetricName.timeout);
+        } else {
+            updateMetricsByEventType(event, analyticsCode, MetricName.err);
+        }
+        return Future.succeededFuture();
+    }
+
+    public <T> void updateMetricsByEventType(T event, String analyticsCode, MetricName result) {
+        if (event instanceof AuctionEvent) {
+            metrics.updateAnalyticEventMetric(analyticsCode, MetricName.event_auction, result);
```

Comment:
Could we resolve metrics name first and then call `updateAnalyticEventMetric(..)` ?